docs/decisions/0001-course-experience.rst

```diff
@@ -0,0 +1,283 @@
+Courses units in LMS
+--------------
+
+Status
+======
+
+Proposed
+
+Context
+=======
+
+NELP business case requires multiples features that are not part
+of open-edx platform core, and it's necessary a new implementation
+that allows users to report and share their opinions about the
+course content and experience during the course.
+
+Decisions
+=========
+
+1. Create a course experience module.
+2. Create ``RateCourse`` model with the following fields.
+
+  * author => Foreign key to user model, the user who set their opinion.
+
+  * status => String field with choices: Liked, Unliked and not-set.
+
+  * course_id => String field, course identifier to allow course rating.
+
+3. Create ``RateSection`` model with the following fields.
+
+  * author => Foreign key to user model, the user who set their opinion.
+
+  * status => String field with choices: Liked, Unliked and not-set.
+
+  * usage_id => String field, this is course subsection identifier.
+
+4. Create ``ReportCourse`` model with the following fields.
+
+  * author => Foreign key to user model, the user who set their opinion.
+
+  * reason => String field with choices:
+
+   - Sexual content
+
+   - Graphic violence.
+
+   - Hateful or abusive content.
+
+   - Copycat or impersonation.
+
+   - Other objection.
+
+  * course_id => String field, course identifier to allow course rating.
+
+5. Create ``ReportSection`` model with the following fields.
+
+  * author => Foreign key to user model, the user who set their opinion.
+
+  * reason => String field with choices:
+
+   - Sexual content
+
+   - Graphic violence.
+
+   - Hateful or abusive content.
+
+   - Copycat or impersonation.
+
+   - Other objection.
+
+  * usage_id => String field, this is course subsection identifier.
+
+6. Set model constrains, each model should be unique per sub-sections and user or course and user.
+7. Create API views for each model.
+
+
+API Examples for sub-sections
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  GET request
+
+  /eox-nelp/api/experience/v1/rate/sections/<usage-id>/
```

Comment:
here the response is filtered by the user so the response would have that form. I mean the status of a specific user.

but with this form, it could return the number of likes of the usage_id(block_id) or that would be another endpoint?

Reply:
first sentence